crates/anvil/src/eth/backend/mem/storage.rs

```diff
@@ -416,13 +416,33 @@ impl MinedTransaction {
         })
     }
 
-    pub fn geth_trace(&self, opts: GethDefaultTracingOptions) -> DefaultFrame {
-        GethTraceBuilder::new(self.info.traces.clone(), TracingInspectorConfig::default_geth())
-            .geth_traces(
-                self.receipt.gas_used().as_u64(),
-                self.info.out.clone().unwrap_or_default().0.into(),
-                opts,
+    pub fn geth_trace(&self, opts: GethDebugTracingOptions) -> GethTrace {
+        let GethDebugTracingOptions { config, tracer, tracer_config, .. } = opts;
+
+        if let Some(GethDebugTracerType::BuiltInTracer(GethDebugBuiltInTracerType::CallTracer)) =
+            tracer
+        {
```

Comment:
let's do a match here because eventually, we want to add the remaining tracers as well

Reply:
hi @mattsse, I am not sure how to handle `PreStateTracer` and `JsTracer` case, seems we need to change `geth_trace` return to an Result type and also add an `inspect` method.

Some pointers would be appreciated! Also I feel a bit overwhelmed to impl this, if you have time, feel free to continue in this branch

Reply:
let's ignore them for now, we'd need to figure out a way to replay transaction first, so let's just leave these match arms empty/return an error